libcudacxx/docs/ptx.md

```diff
@@ -560,6 +560,95 @@ __device__ static inline uint32_t getctarank(
 [`griddepcontrol`]: https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-griddepcontrol
 [`elect.sync`]: https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-elect-sync
 
+
+#### `barrier.cluster`
+
+- PTX ISA: [`barrier.cluster`](https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-barrier-cluster)
+
+**NOTE**: The `.aligned` variants of this instruction are provided for
+completeness only. Their use results in undefined behavior, as C++ compilers
+cannot guarantee that the instruction remains textually aligned throughout the
+compilation process. Using
+`cuda::ptx::barrier_cluster_arrive(cuda::ptx::sem_relaxed, cuda::ptx::aligned)`
+therefore emits a deprecation warning.
```

Comment:
There is a difference in codegen between the aligned and non-aligned versions. If the compiler can prove that the instruction is used in a coalesced code path, then the generated code is the same. When the compiler emits the non-aligned PTX, this can result in the use of a register-valued named barrier, reducing occupancy by a factor of 16. Therefore, even in the face of UB, it can be useful to have the `.aligned` version for debugging purposes.